### ISSUE TITLE
summary plugin migration

### ISSUE THREAD

**Issue by mirekdlugosz** [getpelican/pelican-plugins#1304]
Using instructions from drafted blog post, I managed to port `summary` plugin to namespaced structure.
I keep results of my work on my Github account, in `migration` branch: https://github.com/mirekdlugosz/pelican-plugins-summary/tree/migrate

I have not yet run code styling tools on this repo. I used new version of plugin in my website and confirmed it does not result in regressions on my posts.

**Comment by astrojuanlu** [comment]
Also, this plugin has not yet been uploaded to PyPI, right?

**Comment by mirekdlugosz** [comment]
@astrojuanlu no, it has not. At least not by me. As part of plugin migration, after creating new repository in plugins org, Justin should add PyPI key, so autobot can create and upload package for us.

**Comment by darac** [comment]
Any news on this? The `pelican-plugins-summary` repo says to do `pip install pelican-summary`, but that doesn't exist in PyPI.

**Comment by mirekdlugosz** [comment]
@darac unfortunately, Pelican team never picked it up :shrug: 

[On my website, I use version from git](https://github.com/mirekdlugosz/website/blob/main/requirements.txt#L15):

    https://github.com/mirekdlugosz/pelican-plugins-summary/archive/migrate.zip

**Comment by mirekdlugosz** [comment]
@justinmayer sure, no problem!

I looked into read-more plugin right now, as I am not familiar with it. There is indeed some overlap between them, as they both try to construct summary from the article content. However, `summary` plugin is more flexible. On the surface it seem that these plugins are not compatible with each other, but I have not looked closely.

I see that similar concern about overlap was raised back when `read-more` was being introduced: [pelican-plugins PR #120](https://github.com/getpelican/pelican-plugins/pull/120). Personally, I agree with [kylef that this is responsibility of a theme](https://github.com/getpelican/pelican-plugins/pull/120#issuecomment-31089382). Initial contributor and few people argued that they want "continue reading" link to display on the same line as rest of summary, which is not possible due to theme forcing summary to be in separate block element.

Except that this is easily doable with CSS widely supported since 2011 or so: [codepen link](https://codepen.io/mirekdlugosz/pen/wvVXKMN) (as usual, IE was the last to implement it, but that changed with release of IE9).

If this was my personal project, I would deprecate `read-more` and move with `summary`, because `read-more` does not solve a problem that can't be solved through other means and that sounds like less effort than merging plugins.

On the other hand, it seems that [`read-more` does see some downloads in single-digit range](https://pypistats.org/packages/pelican-read-more), so perhaps there are people who find it useful (or they want to try it out). And I guess it has a benefit of already being ported.

CC @wombelix who I understand to be an active user of `read-more`.
